src/rules/guard-super-call.ts

```diff
@@ -100,7 +119,11 @@ const rule: Rule.RuleModule = {
       if (isSuperHookExpression(node, hook)) {
         errNode = node;
         return true;
-      } else if (node.type === 'IfStatement' && !isSuperHook(node.test, hook)) {
```

Comment:
i think i'd leave this as it is and just change `isSuperHook` to accept an extra `requireTypeCheck` parameter

inside `isSuperHook`, we could then:

```ts
if (requireTypeCheck) {
  return /* whatever a type checked super hook looks like */
}
return /* whatever a regular super hook looks like */
```
```